src/app.rs

```diff
@@ -114,6 +117,10 @@ impl<'a, 'b> Application<'a, 'b> {
         while self.states.is_running() {
             self.timer.restart();
             self.advance_frame();
+            // Enforce maximum FPS
+            while self.timer.elapsed() * self.max_fps < Duration::from_secs(1) {
+                yield_now();
```

Comment:
A) Why don't we use `sleep` here?
B) To ensure we're not using 100% CPU, we should yield / sleep at least once per frame.